src/NServiceBus.Testing.Fakes/TestableIncomingContext.cs

```diff
@@ -14,23 +16,30 @@ public abstract partial class TestableIncomingContext : TestableMessageProcessin
         /// </summary>
         protected TestableIncomingContext(IMessageCreator messageCreator = null) : base(messageCreator)
         {
+
         }
 
         /// <summary>
-        /// A fake <see cref="IBuilder" /> implementation. If you want to provide your own <see cref="IBuilder" /> implementation
+        /// A fake <see cref="IServiceProvider" /> implementation. If you want to provide your own <see cref="IBuilder" /> implementation
         /// override <see cref="GetBuilder" />.
         /// </summary>
-        public FakeBuilder Builder { get; set; } = new FakeBuilder();
+        public IServiceCollection Services { get; set; } = new ServiceCollection();
+
+        IServiceProvider IBehaviorContext.Builder => GetBuilder();
 
-        IBuilder IBehaviorContext.Builder => GetBuilder();
+        IServiceProvider _builder = null;
```

Comment:
I'd propose the following approach instead: https://github.com/Particular/NServiceBus.Testing/pull/197

Reply:
Fine by me